### ISSUE TITLE
Streaming context materializes the visual tower because the family has no text-only skeleton route

### ISSUE THREAD

**Issue by RobTand** [RobTand/prismaquant#872]
## Severity

`P1` — a text-only cost run cannot start on any family whose profile declares
`requires_multimodal_skeleton()`. It took down stage A of the GLM-5.3-Flash
joint-AURA campaign (PrismaBuild action
`dae1474a406ec02ef7a953f8a5d3fb0e1f4a30f99292b00aabc66f424b2c8471`, sparklina,
2026-09-20 18:58:21 UTC, rc 1, attempts 1/1, `retry_safe=false`).

## Evidence at `file:line`

`prismaquant/streaming_model.py:1522-1534` flips `multimodal = True` when the
profile answers `requires_multimodal_skeleton()`:

```python
    if not multimodal:
        from .model_profiles import detect_profile
        _profile = detect_profile(model_path)
        if _profile.requires_multimodal_skeleton():
            print(f"{log_prefix} profile {_profile.name} has no text-only "
                  "skeleton route; using the multimodal construction", flush=True)
            multimodal = True
```

`prismaquant/streaming_model.py:1661` then reads the same variable to decide
whether to load the visual tower off the checkpoint:

```python
    if multimodal and not source_snapshot_only:
        visual_module, visual_prefix = _find_visual_module(model)
        ...
        tensors = _read_layer_to_device(visual_prefix + ".", ...)
```

One flag, two independent questions:

* **construction** — can this family be built text-only at the pinned
  transformers? `Glm5NextProfile.requires_multimodal_skeleton()`
  (`prismaquant/model_profiles/glm5_next.py:218`) answers "no": transformers
  5.16 registers no `Glm5NextForCausalLM`.
* **modality** — will this run drive visual inputs? Only the caller knows.

The first was allowed to imply the second.

## What breaks, and under what inputs

Stage A passes calibrated token IDs. `StreamedCausalLM._prepare`
(`prismaquant/cost_streaming.py:1250-1267`) turns `input_ids` into hidden
states through `base_model.embed_tokens` and nothing else — no `pixel_values`,
no `get_image_features` — so nothing on that path can execute the visual tower.
`prismaquant/cost_streaming.py` contains no reference to `visual` at all.

The run's staged readset is therefore text-only, and the tower load refused
against it:

```
[cost-streaming] profile glm5_next has no text-only skeleton route;
  using the multimodal construction
[cost-streaming] head materialized (3 tensors, rotary re-init) in 2.8s
  File "prismaquant/streaming_model.py", line 1667, in _build_streaming_context
    tensors = _read_layer_to_device(
prismaquant.staged_tier_policy.TierPolicyRefused: staged-tier-forbidden:
  readset-not-staged: /mnt/shared/models/GLM-5.3-Flash-BF16/model-00120-of-00120.safetensors
```

The staged-tier refusal was correct. The read was not. (`120` there is the
checkpoint's total shard count, carried in the refused shard's name — not a
measure of how much vision was read. What was measured is a single
`_read_layer_to_device("model.visual.", ...)` whose first tensor range fell
outside the staged readset.)

Every call site that consumes `ctx.visual_module` already passes
`multimodal=True` explicitly — `sensitivity_probe.py:3788` (visual Fisher
probe, with `visual_requires_grad=True`) and
`incremental_measure_quant_cost.py:1006` (visual cost shard) — so no genuine
visual path depends on the implicit flip.

## Why this is not a tier-policy or measurement problem

The streamed **exporter** already states the correct contract for the same
profile flip: `materialize_tensors_streaming` mirrors
`requires_multimodal_skeleton()` → `stage_multimodal` +
`_skeleton_config_and_class(multimodal=True)` and "the visual tower stays on
meta" (`docs/ARCHITECTURE.md`, "Multimodal-forced export skeleton",
2026-08-27). The streaming context is the one that merged the two axes.

## Fix

Split the two axes inside `_build_streaming_context`: keep `multimodal` as the
construction axis (staging, declared-arch skeleton class, weight map) and gate
the tower load on the caller's own declaration. Explicit `multimodal=True`
keeps its current behaviour byte for byte, including `visual_requires_grad`.

Could I have fixed it? Yes, and did — branch
`fix/glm5next-visual-materialization-20260920`, PR to follow, with a regression
test that fails on `71668b0c87` and passes after.
